src/test/java/de/tum/in/www1/artemis/ExamIntegrationTest.java

```diff
@@ -1696,9 +1710,74 @@ private int getNumberOfProgrammingExercises(Exam exam) {
         return count;
     }
 
-    @Test
+    private void configureCourseAsBonusWithIndividualAndTeamResults(Course course, GradingScale bonusToGradingScale) {
+        ZonedDateTime pastTimestamp = ZonedDateTime.now().minusDays(5);
+        Long courseId = course.getId();
+        TextExercise textExercise = database.createIndividualTextExercise(course, pastTimestamp, pastTimestamp, pastTimestamp);
+        Long individualTextExerciseId = textExercise.getId();
+        TextExercise exerciseWithoutParticipants = database.createIndividualTextExercise(course, pastTimestamp, pastTimestamp, pastTimestamp);
+
+        Exercise teamExercise = database.createTeamTextExercise(course, pastTimestamp, pastTimestamp, pastTimestamp);
+        User student1 = userRepo.findOneByLogin("student1").get();
+        User tutor1 = userRepo.findOneByLogin("tutor1").get();
+        Long teamTextExerciseId = teamExercise.getId();
+        Long team1Id = database.createTeam(Set.of(student1), tutor1, teamExercise, "team1").getId();
+        User student2 = userRepo.findOneByLogin("student2").get();
+        User student3 = userRepo.findOneByLogin("student3").get();
+        User tutor2 = userRepo.findOneByLogin("tutor2").get();
+        Long team2Id = database.createTeam(Set.of(student2, student3), tutor2, teamExercise, "team2").getId();
+
+        // Creating result for student1
+        database.createParticipationSubmissionAndResult(individualTextExerciseId, student1, 10.0, 10.0, 50, true);
+        // Creating 2 results for team1
+        Team team1 = teamRepository.findById(team1Id).get();
+        var result = database.createParticipationSubmissionAndResult(teamTextExerciseId, team1, 10.0, 10.0, 40, true);
+        database.createSubmissionAndResult((StudentParticipation) result.getParticipation(), 50, true);
+
+        // Creating result for student2
+        var student2Result = database.createParticipationSubmissionAndResult(individualTextExerciseId, student2, 10.0, 10.0, 50, true);
+        // Creating result for student3
+        var student3Result = database.createParticipationSubmissionAndResult(individualTextExerciseId, student3, 10.0, 10.0, 30, true);
+        // Creating result for team2
+        Team team2 = teamRepository.findById(team2Id).get();
+        database.createParticipationSubmissionAndResult(teamTextExerciseId, team2, 10.0, 10.0, 80, true);
+
+        // Adding plagiarism cases
```

Comment:
comments like this are fine, as it is not immediately evident what is done here without it